### PR TITLE
Fixes #1298 null value keybindings

### LINKED ISSUE #1298: Method from_instances allows Null values in CIMInstanceName
We do not test for None in key properties when build the instance path from the instance.

DSP0004 states that key properties should not be None.

Suggest adding test to confirm property value is not None and raise error if it is None.

### PR DESCRIPTION
**Pull request by KSchopmeyer**
This is ready for review and merge. See commit for details

### CODE REVIEW

**Comment by andy-maier** [comment]
It seems to me there is a general issue that we (I) don't understand yet, and that applies to all config variables:

It seems that `pywbem.config.XXX` and `pywbem.XXX` are two different objects, even though the names from `pywbem.config` are imported into `pywbem`.

Given that we document to use the config variables in the `pywbem` namespace, but they are documented to be in the `pywbem.config` namespace, this creates a significant chance for users to run into that issue.

We need to do some basic research on this and find a general solution.

**Comment by KSchopmeyer** [comment]
I agree. One more discussion item on the agenda.

**Comment by KSchopmeyer** [comment]
We agreed to change the access methods for config. We will commit this change and andy will do the general change.

File: pywbem/cim_obj.py [hunk 1]
```diff
@@ -253,6 +253,7 @@ class (the parent object) has a list of CIM properties, CIM methods and CIM
 
 from . import cim_xml
 from .config import DEBUG_WARNING_ORIGIN, SEND_VALUE_NULL
+import pywbem.config
```
Comment (andy-maier) [inline pywbem/cim_obj.py]: This should import from `.`, not from `pywbem`, to make sure it remains a relative import. In other words, this would become:
```
from . import config
```
and your new config variable would be accessed as:

```
config.IGNORE_NULL_KEY_VALUE
```

File: pywbem/config.py [hunk 2]
```diff
@@ -93,3 +93,17 @@
 #:
 #: *New in pywbem 0.12.*
 SEND_VALUE_NULL = True
+
+#: Backward compatibility option that controls creating CIMInstanceNames with
```
Comment (andy-maier) [inline pywbem/config.py]: Instead of using plural on a class name, use a link on the class name:
```
... creating :class:`~pywbem.CIMInstanceName` objects with ...

File: pywbem/config.py [hunk 3]
```diff
@@ -93,3 +93,17 @@
...
+#: null values for keybindings.
+
+#: :term:`DSP0004`, section 7.9 specifically forbids key properties with
+#: no values but because pywbem has always allowed this, adding the code to
```
Comment (andy-maier) [inline pywbem/config.py]: Instead of saying "with no values", I suggest to say "that are NULL".

File: pywbem/config.py [hunk 4]
```diff
@@ -93,3 +93,17 @@
...
+#: disallow None as a keybinding value is an incompatible change.
+#:
+#: * True: Pywbem ignores None as a value when keybindings are defined.
```
Comment (andy-maier) [inline pywbem/config.py]: It does not "ignore" None in this case, but "tolerates" it.

File: pywbem/config.py [hunk 5]
```diff
@@ -93,3 +93,17 @@
...
+#: * False (default): Pywbem generates ValueError when keybindings are created
```
Comment (andy-maier) [inline pywbem/config.py]: Use back quotes around True, False, None.

Comment (andy-maier) [inline pywbem/config.py]: It does not "generate" ValueError, but "raises" ValueError. Also, use a link for ValueError.

File: pywbem/config.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbem/config.py]: DONE

File: pywbem/config.py (hunk 3 above)
Comment (KSchopmeyer) [reply pywbem/config.py]: DONE

File: pywbem/config.py (hunk 5 above)
Comment (KSchopmeyer) [reply pywbem/config.py]: DONE
